org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLCompletions.java

```diff
@@ -756,20 +760,47 @@ private void collectCommentCompletion(ICompletionRequest request, ICompletionRes
 	}
 
 	private void collectCharacterEntityProposals(ICompletionRequest request, ICompletionResponse response) {
-		// character entities
-		/*
-		 * int offset = request.getOffset(); Position position = request.getPosition();
-		 * int k = offset - 1; int characterStart = position.getCharacter(); char ch =
-		 * request.getDocument().getText().charAt(characterStart); while (k >= 0 &&
-		 * (Character.isLetter(ch) || Character.isDigit(ch))) { k--; characterStart--; }
-		 * if (k >= 0 && text[k] == '&') { let range =
-		 * Range.create(Position.create(position.line, characterStart - 1), position);
-		 * for (String entity : entities) { if (endsWith(entity, ';')) { String label =
-		 * '&' + entity; result.items.push({ label, kind: CompletionItemKind.Keyword,
-		 * documentation: localize('entity.propose', `Character entity representing
-		 * '${entities[entity]}'`), textEdit: TextEdit.replace(range, label),
-		 * insertTextFormat: InsertTextFormat.PlainText }); } } }
-		 */
+		DOMDocument document = request.getXMLDocument();
+		DOMDocumentType docType = document.getDoctype();
+		if (docType == null) {
+			return;
+		}
+
+		String text = request.getXMLDocument().getText();
+		int offset = request.getOffset();
+		int ampOffset = offset - 1;
+		while (ampOffset >= 0 && Character.isLetterOrDigit(text.charAt(ampOffset))) {

```

Comment:
prefer > -1 to >= 0. That's 1 check (superior) instead of 2 (superior and equal)

Reply:
@fbricon fixed